newrelic/hooks/datastore_firestore.py

```diff
@@ -17,8 +17,49 @@
 from newrelic.common.async_wrapper import generator_wrapper, async_generator_wrapper
 
 
+def _conn_str_to_host(getter):
+    """Safely transform a getter that can retrieve a connection string into the resulting host."""
+
+    def closure(obj, *args, **kwargs):
+        try:
+            return getter(obj, *args, **kwargs).split(":")[0]
+        except Exception:
+            return None
+
+    return closure
+
+
+def _conn_str_to_port(getter):
+    """Safely transform a getter that can retrieve a connection string into the resulting port."""
+
+    def closure(obj, *args, **kwargs):
+        try:
+            return int(getter(obj, *args, **kwargs).split(":")[1])
```

Comment:
Should this be casted to an int? Technically, in the datastore_trace, we defined the type as a string. That being said, it would then have to be casted in the tests to be an int since FIRESTORE_PORT is an integer.

Reply:
I think we can make this a string